h5netcdf/core.py

```diff
@@ -736,6 +747,23 @@ def __init__(self, path, mode="a", invalid_netcdf=None, phony_dims=None, **kwarg
                     % (phony_dims, "sort", "access")
                 )
 
+        # string decoding
+        if h5py.__version__ >= LooseVersion("3.0.0"):
+            if "legacy" in self._cls_name:
+                self.decode_strings = True
```

Comment:
In this case, we should raise an `TypeError` if `decode_strings` was explicitly passed.

Reply:
Calling legacy API with `decode_strings` raises TypeError now.